packages/devtools/lib/src/memory/memory.dart

```diff
@@ -645,120 +724,164 @@ class MemoryScreen extends Screen with SetStateMixin {
       ga.select(ga.memory, ga.inspectClass);
       // User selected a new class from the list of classes so the instance view
       // which would be the third child needs to be removed.
-      _removeInstanceView();
-
-      if (!fromMemoryHover) _resetHistory();
+      removeInstanceView();
 
-      final newTable =
-          row == null ? null : await _createInstanceListTableView(row);
-      _pushNextTable(table, newTable);
+      final InboundsTree inboundTree =
+          row == null ? null : await displayInboundReferences(row);
+      TreeTable<InboundsTreeNode> tree = inboundTree.referencesTable;
+      _pushNextTable(table, tree, inboundTree);
     });
 
     return table;
   }
 
-  Future<Table<InstanceSummary>> _createInstanceListTableView(
+  Future<InboundsTree> displayInboundReferences(
       ClassHeapDetailStats row) async {
-    final table = Table<InstanceSummary>.virtual()
-      ..element.clazz('memory-table');
+    final InboundsTreeData treeData = InboundsTreeData();
+    treeData.data = InboundsTreeNode.root();
+
+    final List<InstanceSummary> instanceRows =
+        await memoryController.getInstances(
+      row.classRef.id,
+      row.classRef.name,
+      row.instancesCurrent,
+    );
 
-    try {
-      final List<InstanceSummary> instanceRows =
-          await memoryController.getInstances(
-        row.classRef.id,
-        row.classRef.name,
-        row.instancesCurrent,
-      );
+    for (var instance in instanceRows) {
+      // Add the instance.
+      final InboundsTreeNode instanceNode = InboundsTreeNode.instance(instance);
+      treeData.data.addChild(instanceNode);
+      instanceNode.addChild(InboundsTreeNode.empty());
+    }
 
-      table.model.addColumn(new MemoryColumnSimple<InstanceSummary>(
-        '${instanceRows.length} Instances of ${row.classRef.name}',
-        (InstanceSummary row) => row.objectRef,
-      ));
-
-      table.model.addColumn(MemoryColumnSimple<InstanceSummary>(
-        '',
-        (InstanceSummary expand) => '<div class="alloc-image"> </div>',
-        cssClass: 'allocation',
-        usesHtml: true,
-        hover: true,
-      ));
-
-      table.model.setRows(instanceRows);
-    } catch (e, st) {
-      framework.toast(
-        'Problem fetching instances of ${row.classRef.name}: $e',
-        title: 'Error',
-      );
-      print('Problem fetching instances of ${row.classRef.name}: $e\n$st');
+    final inboundsTreeTable = InboundsTree(this, treeData, row.classRef.name);
+    inboundsTreeTable.update();
+
+    return inboundsTreeTable;
+  }
+
+  Future<String> computeInboundReference(
+    String objectRef,
+    InboundsTreeNode instanceNode,
+  ) async {
+    final InboundReferences refs = await getInboundReferences(objectRef, 1000);
```

Comment:
[out of scope for this review] Consider renaming to avoid `get*` name.

Also, consider removing the type on the left since it's relatively obvious even though it's not a constructor.